anndata/_core/anndata.py

```diff
@@ -1493,7 +1540,9 @@ def to_memory(self) -> "AnnData":
             mem = backed[backed.obs["cluster"] == "a", :].to_memory()
         """
         if not self.isbacked:
-            raise ValueError("Object is already in memory.")
+            adata, already_in_mem = self._to_memory_copy()
+            if already_in_mem:
+                raise ValueError("Object is already in memory.")
```

Comment:
The tests fail because, in some cases, we expect to_memory to fail when called. So I am not sure if I should check every element and to see if they are dask array or not, just to give an error.

Should the to_memory behavior change, i.e., not give errors anymore

Or should I focus on developing a way to first traverse through the adata to see if it's already on memory, then give an error. This way it would be faster if the call is meant to give an error.

Reply:
I think I'd have it not error anymore. This is what `xarray` does with `load` and `compute`.